lib/compress/index.js

```diff
@@ -1598,6 +1598,38 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every((branch, i) =>
+            (body.length - 1) === i // last branch
```

Comment:
I don't believe we can special case the final branch. Eg,

```js
let i = 1;
switch (id(2)) {
  case 1:
    console.log(1);
    break;
  case 2:
    console.log(2);
    break;
  case ++i:
    console.log(1);
}
```

This shouldn't increment `i`, but the minified output will.

Reply:
oh - yeah that's correct. nice one :)